web/frontend/src/components/camera/camera.svelte

```diff
@@ -1,32 +1,25 @@
 <script lang="ts">
 
-import { onMount, onDestroy } from 'svelte';
+import { onMount } from 'svelte';
 import { displayError } from '@/lib/error';
-import {
-  CameraClient,
-  Client,
-  type ResponseStream,
-  robotApi,
-  type ServiceError,
-} from '@viamrobotics/sdk';
+import { CameraClient, type ServiceError } from '@viamrobotics/sdk';
 import { selectedMap } from '@/lib/camera-state';
-import type { StreamManager } from './stream-manager';
+import { useClient, useDisconnect } from '@/hooks/client';
 
 export let cameraName: string;
-export let client: Client;
 export let showExportScreenshot: boolean;
 export let refreshRate: string | undefined;
-export let streamManager: StreamManager;
-export let statusStream: ResponseStream<robotApi.StreamStatusResponse> | null;
 export let triggerRefresh = false;
 
+const { client, streamManager } = useClient();
+
 let imgEl: HTMLImageElement;
 let videoEl: HTMLVideoElement;
 
 let cameraFrameIntervalId = -1;
 let isLive = false;
 
-const cameraManager = streamManager.setCameraManager(cameraName);
+const cameraManager = $streamManager.setCameraManager(cameraName);
```

Comment:
Does this mutate the `streamManager` value in the store? Is that ok?

Reply:
Yes, and it should be ok since this is really just a readonly store with methods that doesn't change, like the $client store. It's questionable whether this - as well as client - should just be a store or a value returned by `useClient`. I wanted to throw this into the hook as a first decoupling step and then address potential footguns like this afterwards - I'm not particularly a fan of these managers in general and think they can be removed soon-ish